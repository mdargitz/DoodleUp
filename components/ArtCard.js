import React from 'react';
import { connect } from 'react-redux';
import { Text, View, Image, TouchableOpacity, Linking } from 'react-native';
import Loading from './Loading';
import Error from './Error';

function ArtCard(props) {
  if (props.loading || !props.piece) {
    return <Loading />;
  }
  if (props.error) {
    return <Error />;
  }
  return (
    <View style={styles.viewStyle}>
      <TouchableOpacity
        onPress={()=>Linking.openURL(props.piece.url)}
      >
        <Image
          source={{ uri: props.piece.image }}
          style={{ height: 400, width: 300 }}
        />
        <Text style={styles.textStyle}>
          {props.piece.name} by {props.piece.artist}
        </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = {
  viewStyle : {
    justifyContent:'center',
    alignItems: 'center',
    margin: 20,
    backgroundColor: '#fff',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    padding: 30
  },
  textStyle : {
    marginTop: 10
  }
};

const mapStateToProps = (state, props) => {
  return {
    piece: state.inspiration.art[props.number],
    loading: state.inspiration.loading,
    error: state.inspiration.error
  };
};
export default connect(mapStateToProps)(ArtCard);