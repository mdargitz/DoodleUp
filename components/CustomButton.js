import React from 'react';
import { TouchableOpacity, Text, View } from 'react-native';

class CustomButton extends React.Component {
  render() {
    return (
      <View style={styles.viewStyle}>
        <TouchableOpacity 
          style={styles.touchStyle}
          onPress={this.props.onPress}>
          <Text style={styles.textStyle}>{this.props.buttonText}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = {
  viewStyle : {
    alignItems: 'center'
  },
  touchStyle : {
    margin: 8,
    padding: 10,
    backgroundColor: '#bc7c9c',
    borderRadius: 5,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
  textStyle : {
    color: '#fff',
    fontSize: 30,
  }
};

export default CustomButton;