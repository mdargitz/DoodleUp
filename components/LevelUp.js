import React from 'react';
import { Text, View, ScrollView, ImageBackground } from 'react-native';
import { connect } from 'react-redux';
import LevelBtn from './levelup/LevelBtn';

class LevelUp extends React.Component {

  render() {

    return (
      <ScrollView style={{ height: '100%' }}>
        <ImageBackground
          source={{ uri: 'https://image.ibb.co/cJfEwd/mesh_1430107_640.png' }}
          style={{ width: '100%', height: '100%' }}
        >
          <View style={styles.viewStyle}>
            <Text style={styles.headerText}>Complete Challenges to Level Up!</Text>
            <Text style={styles.subtitleText}>You are at level {this.props.level} and currently have {this.props.points} points </Text>
          </View>
          <LevelBtn
            btnText='Level 1 : Form'
            requiredPoints={3}
            requiredLevel={0}
            onPress={() => this.props.navigation.navigate('Level1')}
            image='https://image.ibb.co/hjdZey/sanah_suvarna_161883_unsplash.jpg'
          />
          <LevelBtn
            btnText='Level 2 : Perspective'
            requiredPoints={7}
            requiredLevel={1}
            onPress={() => this.props.navigation.navigate('Level2')}
            image='https://image.ibb.co/dzm1sJ/jorge_gascon_23288_unsplash.jpg'
          />
          <LevelBtn
            btnText='Level 3 : Anatomy'
            requiredPoints={13}
            requiredLevel={2}
            onPress={() => this.props.navigation.navigate('Level3')}
            image='https://image.ibb.co/gEP4CJ/israel_gil_596312_unsplash.jpg'
          />
          <LevelBtn
            btnText='Level 4 : Composition'
            requiredPoints={20}
            requiredLevel={3}
            onPress={() => this.props.navigation.navigate('Level4')}
            image='https://image.ibb.co/dagnzy/shea_rouda_624353_unsplash.jpg'
          />
          <LevelBtn
            btnText='Level 5 : Value &amp; Lighting'
            requiredPoints={30}
            requiredLevel={4}
            onPress={() => this.props.navigation.navigate('Level5')}
            image='https://image.ibb.co/m4B9CJ/reinhart_julian_569049_unsplash.jpg'
          />
        </ImageBackground>
      </ScrollView>
    );
  }
}

const styles = {
  headerText: {
    textAlign: 'center',
    color: '#fff',
    paddingTop: 10,
    fontSize: 30,
    fontWeight: '800'
  },
  subtitleText: {
    textAlign: 'center',
    color: '#fff',
    paddingBottom: 10,
    fontSize: 20
  },
  viewStyle : {
    backgroundColor: 'rgba(0,0,0,.5)'
  }
};

const mapStateToProps = state => {
  return {
    points: state.points.experience,
    level: state.points.level
  };
};

export default connect(mapStateToProps)(LevelUp);