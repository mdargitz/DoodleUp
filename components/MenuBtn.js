import React from 'react';
import { TouchableOpacity, View, Text, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';


class MenuBtn extends React.Component {

  render() {

    let backgroundColor;
    let iconName;
    if (this.props.type === 'Training'){
      backgroundColor = '#BC7C9C';
      iconName = 'line-chart';
    }
    else if (this.props.type === 'LevelUp'){
      backgroundColor = '#A96DA3';
      iconName = 'star';
    }
    else if (this.props.type === 'Inspiration'){
      backgroundColor = '#7A5980';  
      iconName = 'heart';    
    }
    else if (this.props.type === 'Gesture') {
      backgroundColor = '#A96DA3';
      iconName = 'pencil';
    }
    else if (this.props.type === 'Study'){
      backgroundColor = '#7A5980';  
      iconName = 'eye';
    }
    else if (this.props.type === 'FreeDraw'){
      backgroundColor = '#3B3B58';
      iconName = 'paint-brush'; 
    }
    else {
      backgroundColor = '#3B3B58';
      iconName = 'files-o';  
    }

    const styles = StyleSheet.create({
      viewStyle : {
        margin: 20
      },
      touchStyle : {
        backgroundColor,
        justifyContent: 'center',
        alignItems: 'center',
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.8,
        shadowRadius: 2,
        padding: 30
      },
      textStyle : {
        color: '#fff',
        fontSize: 30,
      },
      subtitleStyle : {
        color: '#fff'
      }

    });

    return (
      <View style={styles.viewStyle}>
        <TouchableOpacity
          style={styles.touchStyle}
          onPress={this.props.onPress}
        >
          <Icon name={iconName} size={30} color="#fff" />
          <Text
            style={styles.textStyle}
          >
            {this.props.type}
          </Text>
          <Text
            style={styles.subtitleStyle}
          >
            {this.props.subtitle}
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}




export default MenuBtn;