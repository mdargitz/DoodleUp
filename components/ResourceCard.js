//title text, image thumbnail, short description from parent

import React from 'react';
import { View, TouchableOpacity, Text, Image, Linking } from 'react-native';

export default function ResourceCard(props) {
  return (
    <View>
      <TouchableOpacity
        onPress={() => Linking.openURL('https://www.sophia.org/tutorials/introduction-to-gesture-drawing')}
        style={styles.touchStyle}
      >
        <Image
          source={{ uri: props.image }}
          style={{ width: 200, height: 200 }}
        />
        <Text style={styles.textStyle}>{props.title}</Text>
      </TouchableOpacity>
    </View>

  );
}

const styles = {
  touchStyle : {
    backgroundColor: '#3B3B58',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    padding: 30,
    margin: 20
  },
  textStyle : {
    color: '#fff',
    fontSize: 20,
    padding: 10,
    textAlign: 'center'
  }
};