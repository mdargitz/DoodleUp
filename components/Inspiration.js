import React from 'react';
import { Text, Button, View, Linking, ScrollView, ImageBackground } from 'react-native';
import { fetchArt } from '../actions/inspiration';
import { connect } from 'react-redux';
import ArtCard from './ArtCard';

class Inspiration extends React.Component {

  componentDidMount() {
    this.props.dispatch(fetchArt());
  }
  render() {
    return (
      <ScrollView>
        <ImageBackground
          source={{ uri: 'https://image.ibb.co/cJfEwd/mesh_1430107_640.png' }}
          style={{ height: '100%', width: '100%' }}
        >
          <View style={styles.titleViewStyle}>
            <Text style={styles.headerText}> Check out Some amazing artists</Text>
            <Text style={styles.subtitleText}>Click the links below or take a look at some recent art.</Text>

            <Button
              color='white'
              title='Art Station'
              onPress={() => Linking.openURL('https://www.artstation.com/')}
            />
            <Button
              color='white'
              title='DeviantArt'
              onPress={() => Linking.openURL('https://www.deviantart.com/')}
            />
            <Button
              color='white'
              title='Behance'
              onPress={() => Linking.openURL('https://www.behance.net/')}
            />
          </View>
          <View style={styles.viewStyle}>
            <ArtCard number={0} />
            <ArtCard number={1} />
            <ArtCard number={2} />
            <ArtCard number={3} />
          </View>
        </ImageBackground>
      </ScrollView>
    );
  }
}

const styles = {
  viewStyle: {
    justifyContent: 'center',
  },
  headerText: {
    textAlign: 'center',
    color: '#fff',
    paddingTop: 10,
    fontSize: 30,
    fontWeight: '800'
  },
  subtitleText: {
    textAlign: 'center',
    color: '#fff',
    paddingBottom: 10,
    fontSize: 20
  },
  titleViewStyle: {
    backgroundColor: 'rgba(0,0,0,.5)'
  }
};

export default connect()(Inspiration);