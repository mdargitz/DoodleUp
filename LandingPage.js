import React from 'react';
import { View, Text, ImageBackground } from 'react-native';
import CustomButton from './CustomButton';

export default class LandingPage extends React.Component {
  render() {
    return (
      <ImageBackground
        source={{ uri: 'https://image.ibb.co/ceANzy/background_3497204_640.jpg' }}
        style={styles.imageStyle}
      >
        <View style={styles.viewStyle}>
          <Text style={styles.headerStyle}> Welcome to DoodleUp</Text>
          <Text style={styles.textStyle}> Level Up Your Art </Text>
          <CustomButton
            onPress={() => this.props.navigation.navigate('Home')}
            buttonText='Get Started'
          />
        </View>
      </ImageBackground>
    );
  }
}

const styles = {
  viewStyle: {
    backgroundColor: 'rgba(0,0,0,.5)',
    margin: 20,
    padding: 20
  },
  imageStyle: {
    width: '100%',
    height: '100%',
    justifyContent: 'center'
  },
  headerStyle : {
    fontSize: 50,
    color: '#fff',
    fontWeight: '600',
    textShadowOffset : {width: 2, height: 2},
    textShadowColor: '#000',
    textShadowRadius: 5,
    textAlign: 'center'
  },
  textStyle : {
    fontSize: 20,
    fontWeight: '600',
    color: '#fff',
    textShadowOffset : {width: 2, height: 2},
    textShadowColor: '#000',
    textShadowRadius: 5,
    textAlign: 'center'
  }
};