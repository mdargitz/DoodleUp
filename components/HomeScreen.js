import React from 'react';
import { ScrollView, ImageBackground, Text } from 'react-native';
import MenuBtn from './MenuBtn';

export default class HomeScreen extends React.Component {
  render() {
    const { navigation } = this.props;
    return (
      <ScrollView >
        <ImageBackground
          source={{ uri: 'https://image.ibb.co/g4Gb9y/background_1409028_640.png' }}
          style={styles.imageStyle}
        >
          <MenuBtn
            onPress={() => navigation.navigate('Training')}
            type='Training' 
            subtitle='Refine skills and earn experience points'/>

          <MenuBtn
            onPress={() => navigation.navigate('LevelUp')}
            type='LevelUp' 
            subtitle='Complete challenges to level up'
          />

          <MenuBtn
            onPress={() => navigation.navigate('Inspiration')}
            type='Inspiration'
            subtitle='Check out amazing art around the web' 
          />

          <MenuBtn
            onPress={() => navigation.navigate('Resources')}
            type='Resources' 
            subtitle='Learn with free tutorials and guides'
          />
        </ImageBackground>
      </ScrollView>
    );
  }
}

const styles = {
  imageStyle: {
    width: '100%',
    height: '100%',
  },
  textStyle: {
    color: '#fff',
    fontSize: 30,
    textAlign: 'center',
    padding: 20
  }
};