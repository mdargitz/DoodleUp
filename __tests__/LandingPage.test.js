import React from 'react';
import {LandingPage} from '../components/LandingPage';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<LandingPage />).toJSON();
  expect(tree).toMatchSnapshot();
});