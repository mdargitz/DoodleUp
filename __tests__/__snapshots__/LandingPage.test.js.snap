// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://image.ibb.co/ceANzy/background_3497204_640.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,.5)",
        "margin": 20,
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 50,
          "fontWeight": "600",
          "textAlign": "center",
          "textShadowColor": "#000",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 5,
        }
      }
    >
       Welcome to DoodleUp
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "600",
          "textAlign": "center",
          "textShadowColor": "#000",
          "textShadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "textShadowRadius": 5,
        }
      }
    >
       Level Up Your Art 
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#bc7c9c",
            "borderRadius": 5,
            "margin": 8,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 2,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 30,
            }
          }
        >
          Get Started
        </Text>
      </View>
    </View>
  </View>
</View>
`;
